### DISCUSSION TITLE
Support for Safari

### DISCUSSION THREAD

**Discussion by vaibhav-gupta-haptik** [arnav-kr/json-formatter#20] [Q&A] [answered]
Can we have this extension in the Safari browser as well?
Only reason I'm not using Safari is this extension

**Comment by arnav-kr** [comment, marked as answer]
Actually, The situation is a bit complex.

Theres are several reasons,

Being a student and primary windows/linux user, I don't have a mac and Apple only supports Safari For Macs now.[[1](https://en.wikipedia.org/wiki/Safari_version_history#:~:text=A%20few%20early%20versions%20of%20Safari%20between%203.0%20and%205.1%20were%20released%20for%20Windows.%20Safari%206.0%20and%20on%20are%20only%20available%20on%20macOS%20or%20iOS%20/%20iPadOS.)]
There arises the issue of testing the extension on the actual production browser, Developer experience might be hard in that situation if not impossible.

Porting Extension requires XCode which is again only available for Macs

For both the situations a Mac VM would suffice, but i have a low end laptop which won't be able to run all that.

Assuming even if we find a fix for those, still distribution of a web extension for safari requires to be in Apple Developer Program, whose sign-up requires to have the apple developer app which is only available on Apple devices (there is an alt way to ask the support), and Apple Devloper Program has an annual charge of **$99**[[2](https://developer.apple.com/support/compare-memberships/#:~:text=Enrollment%20is%2099%20USD%20(or%20in%20local%20currency%20where%20available)%20per%20membership%20year.)] they have a fee waiver program but it's only for nonprofit, educational institutions and gov, and i guess json formatter qualifies for none of them.

The annual fee of $99 might be less for some, but currently as a student i can't afford it, also given that i have no income from json formatter (except Ads on the homepage, which i am planing to remove due to aesthetic and authenticity), Supporting Safari currently is not doable from my side, according to current situations, it might be there in future if ever i get a mac/able to pay those charges.

Hope you understand the issues.

Also, would like to thank you for choosing JSON Formatter and supporting it!
